inspirehep/modules/records/receivers.py

```diff
@@ -206,3 +215,45 @@ def enhance_after_index(sender, json, record, *args, **kwargs):
 
     elif is_data(json):
         populate_citations_count(record=record, json=json)
+
+
+@after_record_insert.connect
+@after_record_update.connect
+def after_update(sender, *args, **kwargs):
+    json = kwargs.get('record')
+
+    if json.get('deleted'):
+        # in this way the references diff will be the list of references itself
+        record = InspireRecord(json)
+    else:
+        pid_type = get_pid_type_from_schema(json['$schema'])
+        pid_value = json['control_number']
+        try:
+            rec_uuid = PersistentIdentifier.get(pid_type, pid_value).object_uuid
+            record = InspireRecord.get_record(rec_uuid)
+        except NoResultFound:
+            # deleted record
+            record = InspireRecord(json)
+
+    modified_references = record.get_modified_references()
+
+    pids = [
+        str(get_recid_from_ref(ref['record'])) for ref in modified_references
+    ]
+    uuids = [
+        t[0] for t in db.session.query(PersistentIdentifier.object_uuid)
+        .filter(PersistentIdentifier.pid_type == 'lit')
+        .filter(PersistentIdentifier.pid_value.in_(pids))
+        .all()
```

Comment:
* for `dat` to work too, you need to do something like https://github.com/inspirehep/inspire-next/blob/4d116e13e625dc4ea4a8b9a3ab2bcf6261ecec77/inspirehep/utils/record_getter.py#L132-L135
* `t.object_uuid` is probably clearer than `t[0]`.
* **For later** we should really think hard about how to organize DB queries, currently they happen everywhere and they are often partially duplicated.

Reply:
:+1: